core/src/col_metadata.rs

```diff
@@ -256,24 +256,26 @@ impl ResultSetSchema {
                 .flatten_ok()
                 .collect::<Result<HashMap<Vec<String>, MongoColMetadata>>>()?;
 
-        Ok(match self.select_order {
-            // in the select list order is None, for example if using an older adf version, sort
-            None => processed_result_set_metadata
-                .into_values()
-                .sorted_by(|a, b| match Ord::cmp(&a.table_name, &b.table_name) {
-                    core::cmp::Ordering::Equal => Ord::cmp(&a.col_name, &b.col_name),
-                    v => v,
-                })
-                .collect(),
-            // given a select order, convert the values of the map into an ordered vector
-            _ => self
-                .select_order
-                .as_ref()
-                .unwrap()
-                .iter()
-                .map(|key| processed_result_set_metadata.remove(key).unwrap())
-                .collect(),
-        })
+        Ok(
+            // the select list order is None or empty, for example if using an older adf version, sort by column name
+            if self.select_order.is_none() || self.select_order.as_ref().unwrap().is_empty() {
+                processed_result_set_metadata
+                    .into_values()
+                    .sorted_by(|a, b| match Ord::cmp(&a.table_name, &b.table_name) {
+                        core::cmp::Ordering::Equal => Ord::cmp(&a.col_name, &b.col_name),
+                        v => v,
+                    })
+                    .collect()
+            } else {
+                // given a select order, convert the values of the map into an ordered vector
+                self.select_order
+                    .as_ref()
+                    .unwrap()
+                    .iter()
+                    .map(|key| processed_result_set_metadata.remove(key).unwrap())
```

Comment:
It's probably "safer" here to use `filter_map` and obviate the unwrap.


```suggestion
                    .filter_map(|key| processed_result_set_metadata.remove(key))
```

Reply:
@terakilobyte We are not filtering and `procesed_result_set_metadata` is short lived.
What about doing ' .map(|key| processed_result_set_metadata.get(key))` instead?
It achieves the same purpose and matches the logic.

Reply:
`filter_map` just removes None values, no? It just removes the need to call `unwrap`.

Reply:
ah, I see, I didn't think about the possible case where a key is not present in the result set metadata.
Something would not be quite right if that was the case.
Let me see, we should gracefully error out, not panic, but also not just ignore the missing key